File: go-version/leetcode/LC2270/ways_to_split_array.go
package lc2270

func waysToSplitArray(nums []int) int {
	left, right := int64(0), int64(0)
	for _, num := range nums {
		right += int64(num)
	}

	ans, n := 0, len(nums)
	for i := 0; i < n-1; i++ {
		left += int64(nums[i])
		right -= int64(nums[i])
		if left >= right {
			ans++
		}
	}
	return ans
}

File: python-version/LC2270/ways_to_split_array.py
import unittest

class Solution:
    def waysToSplitArray(self, nums: list[int]) -> int:
        n = len(nums)
        left, right = 0, sum(nums)
        ans = 0
        for i in range(n-1):
            left += nums[i]
            right -= nums[i]
            if left >= right:
                ans += 1
        return ans
    
class SolutionTest(unittest.TestCase):
    def TestWaysToSplitArray(self):
        solution = Solution()
        self.assertEqual(2, solution.waysToSplitArray([10,4,-8,7]))
        self.assertEqual(2, solution.waysToSplitArray([2,3,1,0]))

if __name__ == '__main__':
    unittest.main()